src/nlp/arrow_dataset.py

```diff
@@ -406,6 +443,7 @@ def flatten_(self, max_depth=16):
         logger.info(
             "Flattened dataset from depth {} to depth {}.".format(depth, 1 if depth + 1 < max_depth else "unknown")
         )
+        self._fingerprint = update_fingerprint(self._fingerprint, self.__class__.flatten_, {"max_depth": max_depth})
```

Comment:
Couldn't this almost be a decorator wrapping the class methods?
So you just write `@fingerprint` in front of the methods updating the fingerprint and it uses the transform name and all the arguments?
(not sure about it but it would be easier to maintain)